kafka-rest/src/main/java/io/confluent/kafkarest/common/KafkaFutures.java

```diff
@@ -40,6 +44,7 @@ public static <T> CompletableFuture<T> toCompletableFuture(KafkaFuture<T> kafkaF
           if (exception == null) {
             completableFuture.complete(value);
           } else {
+            log.error("Caught Exception on completion of KafkaFuture", exception);
```

Comment:
nit: Isn't this a bit too aggressive? Some of the exceptions thrown here will be handled correctly by our exception mappers and won't necessarily require a full stack trace dump at log level `ERROR`.
I'd suggest downgrading this all the way to `DEBUG` so that it doesn't go into the logs by default, but could be logged if needed.

Reply:
Yeah, it's an interesting thing to discuss.

IMO mostly for log hygiene reasons.
Logs from production systems tend to very easily become unusable without extensive post-processing due to inconsistent log level usage, inconsistent error logging, and lack of anchoring (i.e. using a shared identifier or anchor for more complex operations or related operations, trace IDs being one example of a type of anchoring). If we happen to log 2 times more than what we would log otherwise, this usually translates to production systems keeping logs for 2 times shorter time windows, and that might also be a problem.

The problem with logging at `DEBUG` level here is that we might not be able to reproduce the problem that threw an exception after we elevate the corresponding log level. I haven't seen too many of these cases with Kafka REST so I'm partially influenced by that.